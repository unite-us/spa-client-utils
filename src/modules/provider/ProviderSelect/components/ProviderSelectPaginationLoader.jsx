import React from 'react';
import PropTypes from 'prop-types';
import _ from 'lodash';
import { Button } from '@unite-us/ui';
import ProviderCardLoading from './ProviderCardLoading';

function hasMorePages(pagingObject) {
  return !!pagingObject.next_page;
}

const ProviderSelectPaginationLoader = ({
  onLoadMore,
  isFetching,
  isReferral,
  paging,
}) => {
  if (isFetching) {
    return (
      <div className="ui-provider-select__load-more--fetching">
        <ProviderCardLoading />
      </div>
    );
  }

  if (hasMorePages(paging) && !_.isEmpty(paging)) {
    const remainingProviders = paging.total_count - (50 * paging.current_page);
    const nextLoadSize = remainingProviders > 50 ? 50 : remainingProviders;
    let label = `Load ${nextLoadSize} More Results`;
    label = isReferral ? 'Load More Results' : label;

    return (
      <Button
        className="ui-provider-select__load-more"
        id="load-more"
        label={label}
        primary
        onClick={onLoadMore}
      />
    );
  }

  return null;
};

ProviderSelectPaginationLoader.propTypes = {
  isFetching: PropTypes.bool.isRequired,
  isReferral: PropTypes.bool.isRequired,
  onLoadMore: PropTypes.func.isRequired,
  paging: PropTypes.object.isRequired,
};

ProviderSelectPaginationLoader.defaultProps = {
  isFetching: false,
  paging: {},
};

export default ProviderSelectPaginationLoader;
