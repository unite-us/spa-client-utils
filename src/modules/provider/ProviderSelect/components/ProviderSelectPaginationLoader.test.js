import renderer from 'react-test-renderer';
import React from 'react';
import ProviderSelectPaginationLoader from './ProviderSelectPaginationLoader';

describe('ProviderSelectPaginationLoader', () => {
  const props = {
    onLoadMore: jest.fn(),
    emptyNextPageMessage: 'We have no more',
    isFetching: false,
    isReferral: false,
    paging: {
      current_page: 1,
      next_page: 2,
      total_count: 100,
    },
  };

  let tree = renderer.create(<ProviderSelectPaginationLoader {...props} />);

  it('renders "Load More"', () => {
    expect(tree).toMatchSnapshot();
  });

  it('renders the empty message', () => {
    tree = renderer.create(<ProviderSelectPaginationLoader {...props} paging={{ next_page: null }} />);
    expect(tree).toMatchSnapshot();
  });

  it('renders the the default empty message', () => {
    tree = renderer.create(<ProviderSelectPaginationLoader {...props} emptyNextPageMessage={undefined} paging={{ next_page: null }} />);
    expect(tree).toMatchSnapshot();
  });

  it('renders null, returns no empty message', () => {
    tree = renderer.create(<ProviderSelectPaginationLoader {...props} emptyNextPageMessage={null} paging={{ next_page: null }} />);
    expect(tree).toMatchSnapshot();
  });

  it('renders html empty message', () => {
    tree = renderer.create(<ProviderSelectPaginationLoader {...props} emptyNextPageMessage={<div>No More</div>} paging={{ next_page: null }} />);
    expect(tree).toMatchSnapshot();
  });

  it('renders the ProviderCardLoading when loading', () => {
    tree = renderer.create(<ProviderSelectPaginationLoader {...props} isFetching />);
    expect(tree).toMatchSnapshot();
  });
});
