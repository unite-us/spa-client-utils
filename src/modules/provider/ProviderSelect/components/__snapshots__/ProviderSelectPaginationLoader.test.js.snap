// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProviderSelectPaginationLoader renders "Load More" 1`] = `
<button
  aria-label="Load 50 More Results"
  className="ui-button ui-button--primary ui-provider-select__load-more"
  data-test-element={null}
  disabled={false}
  id="load-more"
  onClick={[Function]}
  style={Object {}}
  type="button"
>
  Load 50 More Results
</button>
`;

exports[`ProviderSelectPaginationLoader renders html empty message 1`] = `null`;

exports[`ProviderSelectPaginationLoader renders null, returns no empty message 1`] = `null`;

exports[`ProviderSelectPaginationLoader renders the ProviderCardLoading when loading 1`] = `
<div
  className="ui-provider-select__load-more--fetching"
>
  <div
    className="ui-provider-card"
  >
    <div
      className="ui-provider-card__info"
    >
      <div
        className="ui-provider-card__detail"
      >
        <div
          className="bar-loader bar-loader--tall bar-loader--semi-full mb-half"
          id=""
          style={Object {}}
        />
        <div
          className="bar-loader bar-loader--quarter mb-half"
          id=""
          style={Object {}}
        />
        <div
          className="bar-loader bar-loader--half mb-half"
          id=""
          style={Object {}}
        />
        <div
          className="bar-loader bar-loader--half mb-half"
          id=""
          style={Object {}}
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`ProviderSelectPaginationLoader renders the empty message 1`] = `null`;

exports[`ProviderSelectPaginationLoader renders the the default empty message 1`] = `null`;
